R CODE FINAL Alexander Shemetev

```diff
@@ -0,0 +1,791 @@
+
+mis_u1
+bankloans_missing$logit_u_boolean = ifelse(bankloans_missing$logit_u >= 0.5, 1, 0)
+bankloans_missing$logit_u_boolean
+# This (the last line) returns boolean matrix where 1 = threshold value is estimated above and 0 - otherwise
+# These values are attached to the main element as a new column
+# 7) Compute summary statistics for the obtained predictions and for the defaults column of bankloans_existing. Compute the fractions of
+#    defaults (you may use one of the functions you programmed before). Do the results differ a lot?
+describe(bankloans_missing$logit_u)
+describe(bankloans_missing$logit_u_boolean)
+
+# Returns: 
+# bankloans_missing$logit_u 
+#n  missing distinct     Info      Sum     Mean      Gmd 
+#150        0        2    0.403       24     0.16   0.2706 
+# We have 150 observations; 24 are above threshold; others - are below (missing values is 0)
+var_logit_sens_score(bankloans_missing$logit_u_boolean)
+var_logit_sens_score(bankloans_missing$income)
+var_logit_sens_score(bankloans_new$income)
+# No great difference in results - the matrix is similar optimal value in 55 percentile accuracy
+
+# 8) Replace NA's in default columns of bankloans_missing with the predicted values. Append bankloans_missing with bankloans_existing
+bankloans_missing$default = bankloans_missing$logit_u_boolean
+bankloans_missing$default
+# Cleaning the data from the additional data that was used
+bankloans_missing = subset(bankloans_missing, select = -def_exists )
+bankloans_missing = subset(bankloans_missing, select = -dummy_model )
+bankloans_missing = subset(bankloans_missing, select = -logit_u )
+bankloans_missing = subset(bankloans_missing, select = -logit_u_boolean)
+
+total_df = rbind(bankloans_existing, bankloans_missing)
+
+describe(total_df$default)
+# Output
+#total_df$default 
+#n  missing distinct     Info      Sum     Mean      Gmd 
+#850        0        2    0.553      207   0.2435   0.3689 
+# 850 observations; no missing values - missing values replaced with the predictions from our regression
+
+# 9) Save data as bankloans.xlsx with 3 sheets, on the first sheet there should be a dataframe you obtained in part 8 above, sheet 2 
+#    must contain bankloans_existing dataframe, and sheet 3 must contain the transposed version of bankloans_missing dataframe
+install.packages("xlsx")
+library("xlsx")
+write.xlsx(total_df, file = "merry_christmas.xlsx", sheetName = "processed_data", append = FALSE)
+write.xlsx(bankloans_existing, file = "merry_christmas.xlsx", sheetName="without_na", append=TRUE)
+loans_miss_transpose <- as.data.frame(t(as.matrix(bankloans_missing)))
+loans_miss_transpose
+write.xlsx(loans_miss_transpose, file = "merry_christmas.xlsx", sheetName="transposed_d", append=TRUE)
+# The final Excel has all 3 sheets with the prescribed data in
+# Merry Christmas! 
+
```

Comment:
Overall good. You too often create additional unnecessary (sometimes even absolutely unnecessary) variables